greedy-timetabling/src/main/java/org/acme/schooltimetabling/rest/TimeTableResource.java

```diff
@@ -0,0 +1,125 @@
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.acme.schooltimetabling.rest;
+
+import io.quarkus.panache.common.Sort;
+import org.acme.schooltimetabling.domain.Lesson;
+import org.acme.schooltimetabling.domain.Room;
+import org.acme.schooltimetabling.domain.TimeTable;
+import org.acme.schooltimetabling.domain.Timeslot;
+import org.acme.schooltimetabling.persistence.LessonRepository;
+import org.acme.schooltimetabling.persistence.RoomRepository;
+import org.acme.schooltimetabling.persistence.TimeslotRepository;
+
+import javax.inject.Inject;
+import javax.transaction.Transactional;
+import javax.ws.rs.GET;
+import javax.ws.rs.Path;
+import java.util.Objects;
+
+@Path("timeTable")
+public class TimeTableResource {
+    public static final Long SINGLETON_TIME_TABLE_ID = 1L;
+    @Inject
+    TimeslotRepository timeslotRepository;
+    @Inject
+    RoomRepository roomRepository;
+    @Inject
+    LessonRepository lessonRepository;
+
+    // To try, open http://localhost:8080/timeTable
+    @GET
+    public TimeTable getTimeTable() {
+        return findById(SINGLETON_TIME_TABLE_ID);
+    }
+
+    @GET
+    @Path("solved")
+    public TimeTable getTimeSolvedTable() {
+
+        TimeTable timeTable = findById(SINGLETON_TIME_TABLE_ID);
+
+        for (Lesson lesson : timeTable.getLessonList()) {
+            int bestScore = Integer.MIN_VALUE;
+            Room bestRoom = null;
+            Timeslot bestTimeslot = null;
+            for (Room room : timeTable.getRoomList()) {
+                for (Timeslot timeslot : timeTable.getTimeslotList()) {
+                    int score = calculateScore(timeTable, timeslot, room, lesson.getStudentGroup(), lesson.getTeacher());
+                    if (score > bestScore) {
+                        bestScore = score;
+                        bestRoom = room;
+                        bestTimeslot = timeslot;
+                    }
+                }
+            }
+            lesson.setTimeslot(bestTimeslot);
+            lesson.setRoom(bestRoom);
+            timeTable.setScore(timeTable.getScore() + bestScore);
+            timeTable.setSolved();
+        }
+        return timeTable;
+    }
+
+    private int calculateScore(TimeTable unsolved, Timeslot timeslot, Room room, String studentGroup, String teacher) {
+        for (Lesson l : unsolved.getLessonList()) {
+            if (l.getRoom() != null
+                    && l.getTimeslot() != null
+                    && Objects.equals(l.getRoom().getId(), room.getId())
+                    && Objects.equals(l.getTimeslot().getId(), timeslot.getId())
+            ) {
+                return -1;
```

Comment:
let's teach them the right way, right away: quantify the number of hard constraints broken.

So here: hardScore--;
and return the hardScore at the end of this calculateScore method

Reply:
start the calculateScore method with "int hardScore = 0;"

Note that it will calculate the score for all broken hard constraints, not just the ones broken in the current move, which is what we want.

Reply:
In fact the current code is broken for overconstrained problems, or problems that break hard constraints in the greedy algo: once they break one hard constraints, they'll random start taking any move, and stop minimizing the overall hard constraints. The suggested code changes ^^ will fix that.